spec/enumerable_spec.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require_relative '../my_enumerable_module.rb'
+
+# rubocop: disable Metrics/BlockLength, Layout/LineLength
+
+describe Enumerable do
+  describe '#my_each' do
+    it 'Returns the same array if block is given' do
+      expect([5, 6, 7].my_each { |x| x }).to eql([5, 6, 7])
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect([5, 6, 7].my_each.class).to eql(Enumerator)
+    end
+  end
+  describe '#my_each_with_index' do
+    it 'Returns the same array if block is given' do
+      expect([5, 6, 7].my_each_with_index { |x, y| x + y }).to eql([5, 6, 7])
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect([5, 6, 7].my_each_with_index.class).to eql(Enumerator)
+    end
+  end
+  describe '#my_select' do
+    it 'Returns result of the condition inside the block if block is given' do
+      expect([5, 6, 7].my_select { |x| x > 5 }).to eql([6, 7])
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect([5, 6, 7].my_select.class).to eql(Enumerator)
```

Comment:
- [ ] consider putting `[5,6,7]` into a variable so you don't repeat it through out your code. Try the following
```ruby
let(:arr){[5,6,7]}
```
Then use `arr`in your test cases

Reply:
Thank you for the review, we will fix that 👍